src/form/TextInput.tsx

```diff
@@ -5,17 +5,21 @@ import formStyles from "../styles/Forms.module.css";
 interface MyProps {
   title: string,
   name: string,
-  value: string | null,
+  value: string | null | number,
   valueChanged: (e: React.ChangeEvent<HTMLInputElement>) => void,
-  clearInput: (name: string) => void,
+  clearInput?: (name: string) => void,
   validated: boolean,
   errorMessage?: string | false,
   placeholder?: string,
   onFocus?: (e: React.ChangeEvent<HTMLInputElement>) => void,
   onBlur?: () => void,
   handleEnter?: (e: any) => void,
   triedToSubmit?: boolean,
-  readOnly?: boolean
+  readOnly?: boolean,
+  form?: string,
+  type?: string,
+  required?: boolean,
+  showUpDownArrows?: boolean, // only works if type = "number"
```

Comment:
I think if we want to have the upDownArrows then it's better to use the default number input HTML element instead of customising it. It doesn't make sense to have the upDownArrows for a text input component.

Reply:
I will for now not change this, but I think you do have a point.
Not sure about this honestly.

My motivation behind wanting to re-use this TextInput is to be able to re-use the following code.
Might be good if we only have that at one spot.

if (validated) {
        myInput.current.setCustomValidity('');
      } else {
        myInput.current.setCustomValidity(errorMessage || '');
      };

Reply:
Ah I just look at the code again. It's actually a good idea to use this here. It's just a bit confusing that the TextInput field now can sometimes become NumberInput that supports only number. It would be better to rename the field to TextNumberInput in my opinion. However, it's not important, we can just keep it the way it is now.